lib/widget/clubs/club_card.dart

```diff
@@ -0,0 +1,94 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_template/styles/theme_colors.dart';
+import 'package:flutter_template/styles/theme_dimens.dart';
+import 'package:flutter_template/styles/theme_durations.dart';
+import 'package:flutter_template/widget/clubs/animated_gradiant.dart';
+import 'package:flutter_template/widget/provider/data_provider_widget.dart';
+
+class ClubCard extends StatefulWidget {
+  const ClubCard(
+      {Key? key,
+      this.width = 450,
+      this.height = 450,
+      this.title = '',
+      this.url =
+          'https://thumbs.dreamstime.com/b/healthy-lifestyle-fitness-sports-woman-running-sunrise-road-39709089.jpg'})
+      : super(key: key);
+  final double width;
+  final double height;
+  final String title;
+  final String url;
+
+  @override
+  _ClubCardState createState() => _ClubCardState();
+}
+
+class _ClubCardState extends State<ClubCard> {
+  bool isHovered = false;
+  double scaleFactor = 1.1;
+  double currentScale = 1;
+  @override
+  Widget build(BuildContext context) {
+    return DataProviderWidget(childBuilderTheme: (context, theme) {
+      return MouseRegion(
+        cursor: SystemMouseCursors.click,
+        onEnter: (event) {
+          setState(() {
+            currentScale = scaleFactor;
+            isHovered = true;
+          });
+        },
+        onExit: (event) {
```

Comment:
move this to seperate function inside this _ClubCardState. that way your widget tree only contains widgets. and the logic is moved to a function inside this file.